angr/analyses/propagator/propagator.py

```diff
@@ -56,13 +57,18 @@ def merge(self, *others):
     def add_replacement(self, codeloc, old, new):
         """
         Add a replacement record: Replacing expression `old` with `new` at program location `codeloc`.
+        If the self._only_consts flag is set to true, only constant values will be set.
 
         :param CodeLocation codeloc:    The code location.
         :param old:                     The expression to be replaced.
         :param new:                     The expression to replace with.
         :return:                        None
         """
-        self._replacements[codeloc][old] = new
+        if self._only_consts:
+            if isinstance(new, int) or new == TOP:
```

Comment:
There should be only one `TOP` instance ever. Use `is` instead of `==` for comparison.